jena-shex/src/main/java/org/apache/jena/shex/validation/ValidationContext2.java

```diff
@@ -85,6 +85,24 @@ public Graph getGraph() {
         return this.graph;
     }
 
+    public ShexReport dispatchStartSemanticAction(ShexSchema schema, ShexReportElement factory) {
+        ShexReport.Builder builder = new ShexReport.Builder();
+        List<SemAct> semACts = schema.getSemActs();
+        for (SemAct semAct: semACts) {
+            String semActIri = semAct.getIri();
+            SemanticActionPlugin semActPlugin = this.semActPluginIndex.get(semActIri);
+            if (semActPlugin != null) {
+                if (!semActPlugin.evaluateStart(semAct, schema)) {
+                    ShexReportElement r = ExhaustiveShexReportElement.factory(); // the passed ShexReportElement factory doesn't accept create(null, null)
+                    r.setSatisfies(false);
+                    r.addInfoFailure(String.format("%s start shape failed", semActIri), null, null);
+                    builder.addReport(null, null, r);
```

Comment:
The `node` and `shapeLabel` are null because the startSemActs are invoked before diving into the ShapeMap.